src/app/auth.service.spec.ts

```diff
@@ -0,0 +1,131 @@
+import {TestBed, inject} from '@angular/core/testing';
+import {AuthService} from './auth.service';
+
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+
+import {AngularFireModule} from '@angular/fire';
+import {AngularFireAuth} from '@angular/fire/auth';
+import {auth} from 'firebase/app';
+
+import {Recommendation} from './models/recommendation.model';
+import {environment} from '../environments/environment';
+
+describe('AuthService', () => {
+  let service: AuthService;
+
+  const credential = {
+    user: {},
+    credential: {},
+  };
+
+  const idToken = 'idT0ken';
+
+  /* TODO Most iffy about currentUser */
+  const mockAngularFireAuth = jasmine.createSpyObj('AngularFireAuth', {
+    signInWithPopup: Promise.resolve(() => {
+      return credential;
+    }),
+    signOut: Promise.resolve(() => {}),
+    currentUser: Promise.resolve(user => {
+      return jasmine.createSpyObj('user', {
+        getIdToken: Promise.resolve(() => {
+          return idToken;
+        }),
+      });
+    }),
+  });
+
+  const mockWindow = {
+    location: jasmine.createSpyObj('location', ['reload']),
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        AngularFireModule.initializeApp(environment.firebaseConfig),
+      ],
+      providers: [
+        AuthService,
+        {provide: Window, useValue: mockWindow},
+        {provide: AngularFireAuth, useValue: mockAngularFireAuth},
+      ],
+    });
+    service = TestBed.inject(AuthService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it(`should authenticate user with Google pop-up and return user's ID token`, () => {
+    /* TODO
+       [ ] ?. Check that user isn't signed in, i.e. currentUser is null
+       [ ] ?. Check that getIdToken isn't on currentUser (because null)
+       [x] 1. Authenticate user via service.authenticateWithGoogle()
+       [x] 2. Check that user was prompted with Google auth
+       [ ] 3. Check that currentUser doesn't resolve to a null value
+       [x] 4. Check that currentUser.getIdToken was called
+       [ ] 5. Check that currentUser.getIdToken is not null
+    */
```

Comment:
for the sake of this test, I don't think you need to mock currentUser.getIdToken because the functionality of authenticateWithGoogle() doesn't depend on it being set.

it could make sense (in a separate `it('should fail gracefully if signing in does not set currentUser')`) to test the case where `user === null` in the callback.